### PR TITLE
Basic Scalar, Matrix and Vector Math

### PR DESCRIPTION
**Pull request by RealNeGate**

### CODE REVIEW

**Review by netizen-a — CHANGES_REQUESTED** [review]
add a `static_assert` to all new functions in `Mat4x4.h` to showcase their functionality. Make the functions `contexpr` if needed.

**Review by netizen-a — APPROVED** [review]
Looks good the basic math functions can be converted to intrinsics now.

File: src/Boxer/Common.h [hunk 1]
```diff
@@ -40,10 +44,17 @@ DeferInternal<F> defer_func(F f) {
...
+#else
+// IDK
+#define INLINE
+#endif
+
 // TODO(NeGate): Clean up!
 inline void AssertFunc(bool condition, const char* err = nullptr) {
 	if(!condition) {
-		printf("Fuck! %s\n", (err != nullptr) ? err : "");
+		printf("Darn! %s\n", (err != nullptr) ? err : "");
```
Comment (travisvroman) [inline src/Boxer/Common.h]: Change this to something meaningful. :P

File: src/Boxer/main.cpp [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #ifdef PLATFORM_WINDOWS
 #include <crtdbg.h> // include to bring _CrtSetDbgFlag into scope
 #endif
+#include "Math/Math.h"
+
```
Comment (travisvroman) [inline src/Boxer/main.cpp:10]: ...Do I want to know why main.cpp is including math? It definitely should not be.

**Review by travisvroman — CHANGES_REQUESTED** [review]

**Review by travisvroman — APPROVED** [review]

**Pull request merged**